src/main/java/no/uib/inf112/core/map/GameMap.java

```diff
@@ -62,29 +74,47 @@ public GameMap(String map) {
             throw new IllegalStateException("Given tiled map does not have a tile layer named '" + BOARD_LAYER_NAME + "'");
         }
         if (flags == null) {
-            throw new IllegalStateException("Given tiled map does not have a tile layer named '" + FLAG_LAYER_NAME + "'");
+            lasers = new TiledMapTileLayer(mapWidth, mapHeight, tileWidth, tileHeight);
```

Comment:
Why new laserlayer if one of the layers are null?

Reply:
This is @kapir code

Reply:
Its not